### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by francescofarina**
## Proposed changes

Adding contributing guidelines in CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Contributing guidelines
+
+## Opening an Issue
+
+- Before [creating an issue](https://help.github.com/en/github/managing-your-work-on-github/creating-an-issue), check if you are using the latest version of the project. If you are not up-to-date, see if updating fixes your issue first.
+- Do not to open duplicate issues**; before creating a new issue, go through existing ones to see if it has previously been reported. If your issue exists, comment with any additional information you have.
+- Select the appropriate issue template and fully complete it. If no template is available for your specific issue, please use a blank one and report as much details as needed
```
Comment (TristanBilot) [inline CONTRIBUTING.md:7]:
```suggestion
- Select the appropriate issue template and fully complete it. If no template is available for your specific issue, please use a blank one and report as much details as needed.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Another great way to contribute is to open feature requests!
+While all requests are valid, we cannot guarantee your request will be accepted. Your idea may be great, but also out-of-scope for the project.
+For every feature request, you are welcome to submit a pull request to help!
+
+
+## Asking Questions
+Please use [Discussions](https://github.com/mlx-graphs/mlx-graphs/discussions) to ask questions instead of issues.
+GitHub issues are not the appropriate place to debug your specific project, but should be reserved for filing bugs and feature requests.
+
```
Comment (TristanBilot) [inline CONTRIBUTING.md:23]:
```suggestion
To chat and share random questions with the community, we recommend to join our [Discord server](https://github.com/mlx-graphs/mlx-graphs/discussions/107).

```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+## Pull Requests
+When you submit a pull request, you'll be presented with a template to fill with a description of your contribution and checks on whether you added tests and documentation. Please fill all the parts.
+To open a PR:
+- Fork and submit pull requests to the repo.
```
Comment (TristanBilot) [inline CONTRIBUTING.md:27]:
```suggestion
- Fork the repo and follow the [local installation guidelines](https://github.com/mlx-graphs/mlx-graphs?tab=readme-ov-file#contributing).
- Add your contribution and submit a pull request to the repo.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- If you've added code that should be tested, add tests.
+- Update the necessary documentation
```
Comment (TristanBilot) [inline CONTRIBUTING.md:29]:
```suggestion
- Update the necessary documentation.
```

**Review by TristanBilot — CHANGES_REQUESTED** [review]

**Comment by TristanBilot** [comment]
Nice, initiative! Just proposed some adjustments with useful links.

File: CONTRIBUTING.md (hunk 2 above)
Comment (francescofarina) [reply CONTRIBUTING.md:24]: added the actual link

**Review by TristanBilot — APPROVED** [review]

**Pull request merged**